### DISCUSSION TITLE
How to activate renv in notebook

### DISCUSSION THREAD

**Discussion by rbudnar** [posit-dev/positron#4687] [Q&A] [answered]
Hi all! 

My apologies if this is answered elsewhere or if I'm just missing something obvious (I'm a bit of a newcomer to R as well, coming from python). I was testing out using Jupyter notebooks in positron and I was wondering if it is possible to use the local renv for packages. Renvs work great in R scripts without me having to do anything, but in a notebook context it doesn't find my renv-installed packages (where my notebook is under the `src` folder). After messing around for a while, this looks like it worked inside my notebook:

```
setwd("..") # need to go up a directory first, otherwise renv::activate will create a new renv in the current folder
renv::activate()
.libPaths(renv::paths$library())
```
My question is - is there a better way to do this? Messing around with paths can get clunky, especially if I need to move things around.

Thank you!

**Comment by petetronic** [comment, marked as answer]
Hi @rbudnar, thanks for raising this. I think is expected right now because we set the working directory for the notebook to the folder it is in, so in your case `/src`. If your renv is in the root folder above that, it would not see it.

We could, perhaps, try to detect this situation in projects and special case notebook sessions with env information.

**Reply by petetronic** [reply]
I logged https://github.com/posit-dev/positron/issues/4695 to track this enhancement

**Reply by rbudnar** [reply]
Great, thank you!

**Reply by juliasilge** [reply]
If you are a newcomer to R @rbudnar you might consider not embracing the renv lifestyle right away. Because of the dramatic differences in packaging infrastructure, that kind of tooling is less important than in Python. Instead of being table stakes for healthy functioning (the situation in Python), it is more like a specialized tool that people with very high reproducibility needs use. I think renv is great but it does not occupy the same space that, say, virtual environments do in the Python ecosystem.

**Reply by rbudnar** [reply]
Thanks! Coming from python and other programming languages, it is common to have dependencies isolated for separate projects so I assumed that using renv would be the more robust approach. I also tend to structure projects (albeit sometimes a bit loosely) as shown in [cookie cutter data science](https://cookiecutter-data-science.drivendata.org/#directory-structure), where `src` and `notebooks` are separate top level folders.  Of course these are just holdovers from other languages and not hard requirements, so I'm happy to adjust.  

Thanks for the guidance!

**Reply by juliasilge** [reply]
I have seen project templates like those and think they are super helpful! For most typical data science projects in R, the main difference would be to replace `requirements.txt`, `pyproject.toml`, and `setup.cfg` with a `DESCRIPTION` file to specify what packages you use (that's a great way [to specify dependencies even for non-package projects](https://usethis.r-lib.org/reference/use_description.html)).
